cmd/rokka/cli/copyall/copyall.go

```diff
@@ -0,0 +1,128 @@
+package copyall
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/rokka-io/rokka-go/rokka"
+)
+
+//Options general options for the whole copy-all command
+type Options struct {
+	SourceOrganization      string
+	DestinationOrganization string
+	DryRun                  bool
+	Concurrency             int
+}
+
+type listReturn struct {
+	Cursor     string
+	ItemsCount int
+}
+
+//CopyResult the results of the copy operation
+type CopyResult struct {
+	RokkaHash string
+	Error     error
+}
+
+//StartWorkers starts Copy Workers
+func StartWorkers(options Options, client *rokka.Client, images chan string, results chan CopyResult, quit chan bool) {
+	waitGroup := sync.WaitGroup{}
+	waitGroup.Add(options.Concurrency)
+
+	// Start workers for image copy
+	for i := 0; i < options.Concurrency; i++ {
+		go func() {
+			defer waitGroup.Done()
+			copyWorker(client, images, results, quit, options.SourceOrganization, options.DestinationOrganization, options.DryRun)
+		}()
+	}
+
+	// Start a go-routine to to close the result channel as soon as all workers are done
+	go func() {
+		waitGroup.Wait()
+		close(results)
+	}()
+
+}
+
+func copyWorker(
+	client *rokka.Client,
+	imageFiles chan string,
+	results chan CopyResult,
+	quit chan bool,
+	sourceOrg string,
+	destinationOrg string,
+	dryRun bool,
+) {
+	for hash := range imageFiles {
+		result := CopyResult{
+			RokkaHash: hash,
+		}
+
+		if dryRun {
+			result.RokkaHash = "DRY-RUN-HASH"
```

Comment:
wouldn't you want the RokkaHash to stay the same and here just instead of if/else do `if !dryRun`?

Reply:
I'm not sure that even is needed, for debugging is fine, but apart from that, there's not much point. But your point is valid